### ISSUE TITLE
Pick utility - error on excess values

### ISSUE THREAD

**Issue by nitzcard** [microsoft/TypeScript#56830]
Using Pick on a function argument and then giving it a defined variable with excess properties do not give error like if I write it inline. 

### 🕗 Version & Regression Information
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about

### ⏯ Playground Link

https://www.typescriptlang.org/play?ssl=22&ssc=1&pln=1&pc=1#code/FASwdgLgpgTgZgQwMZQAQEFUG9itQgBwIBsoAuVMAVwFsAjWAbl1ToTHYQuvqeAF9gwCAE8CaAAogkAaygATVAF5UU2QB50AGlQByNhw66AfM2BwqYJBBAB7MKiQALKLIAUCGAHMAzhTVy8gCU2ILAAPThqAC0sXHxCYlJySkJwEj2PhCoAG7K2KycHBQAjDqEJOSoJaj8zM6uMm5YhYZc1eVEpKW1IZGoAOq2MDI++GPg0GDyCuku7jkhqP0+TrZUxIpeIDlo7KiwMMOoxCBy+HS2uxFRqXf3D-FC-UMjYwgTkFDTsxlgWfh-NJAvkcHgKt09HRiAhdFoWMN2F4qgAOeFhFgsIA

### 💻 Code

```ts

interface A {
  apple: number;
  banana: number;
}

type Picked = Pick<A, 'banana'>;

function check(args: Picked) {}

// --------------------------------------------------
const v = { banana: 1, apple: 1 };
check({ banana: 1, apple: 1 }) // Works as intended
check(v)  // should give an error like above
// --------------------------------------------------

// Works as intended
const a: Picked = {
  apple: 'bla',
  orange: 8,
}


```

### 🙁 Actual behavior

check(v) do not raise an error

### 🙂 Expected behavior

check(v) should raise an error

**Comment by Andarist** [comment]
This is working as intended. Object types are not sealed in TypeScript and the excess property check is more like a lint rule than a strong type error. See the docs: https://www.typescriptlang.org/docs/handbook/2/objects.html#excess-property-checks

**Comment by nmain** [comment]
`Pick<...>` has no impact here and the results are exactly the same if you write:

```typescript
type Picked = { banana: number };
```

**Comment by fatcerberus** [comment]
There’s a reason the error message specifies “**Object literal** can only specify known properties”; `v` is not an object literal.  You’d need #12936 to enforce this across the board.
